pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/protobuf/CodecParseMethodGenerator.java

```diff
@@ -293,23 +288,30 @@ private static void generateFieldCaseStatement(final StringBuilder sb, final Fie
         } else if (field.type() == Field.FieldType.MESSAGE){
             sb.append("""
 						final var messageLength = input.readVarInt(false);
-						final var limitBefore = input.limit();
-						// Make sure that we have enough bytes in the message
-						// to read the subObject.
-						// If the buffer is truncated on the boundary of a subObject,
-						// we will not throw.
-						final var startPos = input.position();
-						if ((startPos + messageLength) > limitBefore)
-						    throw new BufferOverflowException();
-						input.limit(startPos + messageLength);
-						final var value = $readMethod;
-						// Make sure we read the full number of bytes. for the types
-						if ((startPos + messageLength) != input.position()) {
-							throw new BufferOverflowException();
+						final $fieldType value;
+						if (messageLength == 0) {
+							value = $fieldValue;
+						} else {
+							final var limitBefore = input.limit();
+							// Make sure that we have enough bytes in the message
+							// to read the subObject.
+							// If the buffer is truncated on the boundary of a subObject,
+							// we will not throw.
+							final var startPos = input.position();
+							if ((startPos + messageLength) > limitBefore)
+								throw new BufferOverflowException();
```

Comment:
BufferUnderflowException? It's parsing code, it means there is not enough bytes to read, which is clearly an underflow.

Reply:
Fixed